contrib/scripts/verifier_diff.py

```diff
@@ -0,0 +1,196 @@
+
+    Returns:
+        None
+    """
+    df = pd.read_csv(os.path.join(output_dir, "insns_diff.csv"))
+
+    if not plot_empty:
+        df = df[df["Insns_Diff"] != 0]
+        df = df[~((df["Insns_Before"] == 0) & (df["Insns_After"] == 0))]
+
+    for test_name, test_df in df.groupby("Test"):
+        plt.figure(figsize=(12, 8), constrained_layout=True)
+        colors = test_df["Insns_Diff"].apply(lambda x: "green" if x > 0 else "red")
+        bars = plt.barh(test_df["Function"], test_df["Insns_Diff"], color=colors)
+
+        for bar, insn_count in zip(bars, test_df["Insns_After"]):
+            width = bar.get_width()
+            xpos = width + 1 if width >= 0 else width - 1
+            ha = "left" if width >= 0 else "right"
+            plt.text(
+                xpos,
+                bar.get_y() + bar.get_height() / 2,
+                f"({insn_count})",
+                va="center",
+                ha=ha,
+                fontsize=8,
+                color="black"
+            )
+
+        plt.xlabel("Verifier Instructions Difference (After - Before)\n(Instructions After shown in brackets)")
+        plt.ylabel("Function Name")
+        plt.title(f"Program and Config: {test_name}")
+
+        filename = test_name.replace("/", "_") + ".png"
+        chart_path = os.path.join(output_dir, filename)
+        plt.savefig(chart_path)
+        plt.close()
+
+        logging.info(f"Saved chart: {filename}.")
+
+    logging.info(f"All charts successfully saved.")
+
+
+def setup_output_dir(file_after: str, file_before: str) -> str:
+    """
+    Generates a unique output directory name using input file names and current timestamp.
+    In addition, stores a copy of the log files in the new directory.
+
+    Args:
+        file_after (str): Path to the "after" verifier log.
+        file_before (str): Path to the "before" verifier log.
+
+    Returns:
+        str: Name of the new output directory.
+    """
+    base_after = os.path.basename(file_after).replace('.', '_')
+    base_before = os.path.basename(file_before).replace('.', '_')
+    timestamp = datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
+    output_dir = f"insn-diff-{base_after}-wrt-{base_before}-{timestamp}"
+    
+    os.makedirs(output_dir, exist_ok=True)
+    logging.info(f"Output directory {output_dir} created.")
+    
+    shutil.copy(file_before, output_dir)
+    shutil.copy(file_after, output_dir)
+    logging.info("Log files successfully backed up.")
+    return output_dir
+
+
+def main():
+    """
+    Entry point for the script. Parses arguments, sets up logging,
+    and runs comparison and visualization.
+    """
+    parser = argparse.ArgumentParser(
+        description="Compare eBPF number of verified instructions from verifier logs."
+    )
+    parser.add_argument("--file-before", required=True, help="Path to the log before patch")
+    parser.add_argument("--file-after", required=True, help="Path to the log after patch")
```

Comment:
nit: As a matter of personal preference, I would make one super minor change to just accept the before / after log files as positional parameters to avoid having to type out `--file-before` and `--file-after`. But it's also fine as is.